org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServers.java

```diff
@@ -80,7 +81,14 @@
 		final CompletableFuture<@NonNull List<T>> init = CompletableFuture.completedFuture(new ArrayList<T>());
 		return executeOnServers(fn).reduce(init, LanguageServers::add, LanguageServers::addAll)
 			// Ensure any subsequent computation added by caller does not block further incoming messages from language servers
-			.thenApplyAsync(Function.identity());
+			.thenApplyAsync(Function.identity())
+			.exceptionally(e -> {
+				if (isRequestCancelledException(e)) {
+					return Collections.emptyList();
+				}
```

Comment:
I don't think it's a good pattern to try to hide the cancellation or other exceptions to clients by default; particularly if we want to implement proper cancellation later: in some cases, a cancelled future should cause a retrigger the computation with updated parameters (eg document changed in between).

Also, as mentioned in other tickets, we'll most likely want the cancellation exception to be caught and propagate to call `.cancel()` on the parent futures.

Reply:
@mickaelistria , yes, I agree with you, I have adapted the PR accordingly and undo a previous mistake that I did by putting logic in `LanguageServers`. Could you take a look again?